### PR TITLE
Create memory traffic generator afu test application

### PR DESCRIPTION
**Pull request by kyrob**
Add a test application to stimulate OFS memory traffic generator AFU configuration space. This will be the memory KPI measurement tool for AC/OSC

### CODE REVIEW

**Review by r-rojo — CHANGES_REQUESTED** [review]
Can you add a meaningful description to this pull request?

File: samples/mem_tg/mem_tg.cpp [hunk 1]
```diff
@@ -0,0 +1,37 @@
+// Copyright(c) 2020-2021, Intel Corporation
```
Comment (tswhison) [inline samples/mem_tg/mem_tg.cpp:1]: (c) date to 2022.

File: samples/mem_tg/mem_tg.h [hunk 2]
```diff
@@ -0,0 +1,205 @@
+// Copyright(c) 2020-2021, Intel Corporation
```
Comment (tswhison) [inline samples/mem_tg/mem_tg.h:1]: (c) to 2022.

File: samples/mem_tg/mem_tg.h [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+
+public:
+  uint32_t count_;
+  uint32_t mem_ch_;
+  uint32_t loop_;
+  uint32_t wcnt_;
+  uint32_t rcnt_;
+  uint32_t bcnt_;
+  uint32_t stride_;
+  uint32_t mem_speed_;
```
Comment (tswhison) [inline samples/mem_tg/mem_tg.h:184]: Probably want to init these in the constructor, even if to 0.

File: samples/mem_tg/tg_test.h [hunk 4]
```diff
@@ -0,0 +1,196 @@
+// Copyright(c) 2020-2021, Intel Corporation
```
Comment (tswhison) [inline samples/mem_tg/tg_test.h:1]: (c) to 2022

File: samples/mem_tg/tg_test.h [hunk 5]
```diff
@@ -0,0 +1,196 @@
...
+        auto ret = config_input_options();
+        if (ret != 0) {
+            std::cerr << "Failed to configure TG input options" << std::endl;
+            return ret;
+        }
+
+        int status = 0;
+        status = run_mem_test();
+
+        return status;
```
Comment (tswhison) [inline samples/mem_tg/tg_test.h:186]: Or just
```c++
return run_mem_test();
```

File: opae.spec.fedora [hunk 6]
```diff
@@ -326,6 +326,7 @@ done
 %{_bindir}/pac_hssi_config.py
 %{_bindir}/rtl_src_config
 %{_bindir}/vabtool
+%{_bindir}/mem_tg
```
Comment (tswhison) [inline opae.spec.fedora:329]: I think maybe your branch is out-of-date with respect to the current master.  Could you rebase it?  The structure of this file, in particular, has changed last Friday.

**Review by tswhison — APPROVED** [review]
thanks!

**Review by tswhison — APPROVED** [review]

**Review by r-rojo — APPROVED** [review]

**Pull request merged**